File: Self_Solving_Algorithm/Cube_Algorithms/Common_Moves/Sexy_Move.py
from ...Cube_Turning.Cube_Turning_CCW import CubeArrayRCCW, CubeArrayUCCW
from ...Cube_Turning.Cube_Turning_CW import CubeArrayRCW, CubeArrayUCW
import copy

SexyMoveNotation = [
    CubeArrayRCW,
    CubeArrayUCW,
    CubeArrayRCCW,
    CubeArrayUCCW
]

def SexyMoveAlgorithm(cube):
    CurrentState = copy.deepcopy(cube)
    for move in SexyMoveNotation:
        CurrentState = move(CurrentState)
    return CurrentState

File: Algorithm/cubeAlgorithms/commonMoves/sexyMove.js
// Sexy move is the commonly seen sequence (R U R' U')

import { CubeArrayRCCW, CubeArrayUCCW } from "../../cubeTurningCCW";
import { CubeArrayRCW, CubeArrayUCW } from "../../cubeTurningCW";

var SexyMoveNotation = [
  CubeArrayRCW,
  CubeArrayUCW,
  CubeArrayRCCW,
  CubeArrayUCCW,
];

export function SexyMoveAlgorithm(cube) {
  let CurrentState = JSON.parse(JSON.stringify(cube));
  for (let i = 0; i < SexyMoveNotation.length; i++) {
    CurrentState = SexyMoveNotation[i](CurrentState);
  }

  return CurrentState;
}
